### PR TITLE
Fuse gates referenced by pointers.

### PR DESCRIPTION
**Pull request by sergeisakov**

### CODE REVIEW

File: lib/fuser_basic.h [hunk 1]
```diff
@@ -26,10 +26,26 @@ namespace qsim {
...
 struct BasicGateFuser final {
-  using GateFused = qsim::GateFused<Gate>;
+ private:
+  using RGate = typename std::remove_pointer<Gate>::type;
+
+  static const RGate& ConstRefToGate(const RGate& gate) {
+    return gate;
+  }
+
+  static const RGate& ConstRefToGate(const RGate* gate) {
```
Comment (95-martin-orion) [inline lib/fuser_basic.h:43]: Should these be `GateToConstRef` instead? It looks like the output is a const ref, rather than the input.

**Review by 95-martin-orion — APPROVED** [review]
This all looks good - is there an upcoming feature that this is meant to enable?

**Comment by sergeisakov** [comment]
> This all looks good - is there an upcoming feature that this is meant to enable?

Yes, this will be used in the upcoming quantum trajectory simulator.

File: lib/fuser_basic.h (hunk 1 above)
Comment (sergeisakov) [reply lib/fuser_basic.h]: Replaced `ConstRefToGate` with `GateToConstRef`.

Comment (sergeisakov) [reply lib/fuser_basic.h]: Replaced `ConstRefToGate` with `GateToConstRef`.

**Pull request merged**